spec/definitions/DashboardRequest.yaml

```diff
@@ -0,0 +1,19 @@
+type: object
+properties:
+  segments:
+    type: object
+    description: Filtered segments for each metric
+    properties:
+      segmentName:
```

Comment:
```
{
    segments: {
        'Sales Value': {
            metric: 'salesValue',
            criteria: null
        },
        'Approval Rate': {
            metric: 'approvedThroughput',
            criteria: null
        }
    }
}
```
@adamaltman as I understood your proposition, I made everything a segment, but there is an issue (solvable with sort of prefixes): if we want a subsegment `facebook`, we need to somehow deal with the name:
```
{
    segments: {
        'Sales Value': {
            metric: 'salesValue',
            criteria: null
        },
        'Approval Rate': {
            metric: 'approvedThroughput',
            criteria: null
        }
        'facebook-salesValue': {
            metric: 'salesValue',
            criteria: {somecriteria}
        },
        'facebook-approvedThroughput': {
            metric: 'approvedThroughput',
            criteria: {somecriteria}
        }
    }
}
```

Reply:
I think this design is problematic. Let's discuss it on slack in more detail.  It violates many api design principles.